### DISCUSSION TITLE
opcua client loses connection after restart opcua server

### DISCUSSION THREAD

**Discussion by pfournea** [eclipse-milo/milo#1303] [General]
Hi,

I have read that the opcuaclient always tries to reconnect if a connection is lost , but I my case I don't see this happening.

Scenario:

- Startup Spring Boot application  - creating a opcua client singleton bean
- Read values  (works fine!)
- Shutdown opcua server 
- Read values (got error)
- Startup opcua server
- Read values (still got same error)

After shutdown of the opcua server I receive the answer below while reading the value of a node:

`UaException: status=Bad_Timeout, message=requestId=9533 timed out after 60000ms`

After restart of the opcua server I still receive the same exception while reading the value of a specific node

`UaException: status=Bad_Timeout, message=requestId=11182 timed out after 60000ms`

Only after a restart of my application (and the recreation of the opcuaClient) the issue is resolved and reading goes fine again

I don't  use subscriptions, reading is done by:

```
var opcNodeId = new NodeId(nameSpaceIndex, tagName.getTagValue());
var opcNode = client.readValue(Double.MAX_VALUE, TimestampsToReturn.Server, opcNodeId).get(
                    maxSecondsToWait,
                    TimeUnit.SECONDS
                );
```

Do I need to handle this situation in my code self (re-creation of the opcua client of timeouts are happening)?

Thx for the advice!

Kind Regards,

Peter

**Comment by kevinherron** [comment]
You shouldn't need to. Maybe you can take a Wireshark capture while you go through the steps you subscribed.

Depending on whether the server shutdown is "clean" or not the client may not detect that it's offline until the application layer keep alive reads have failed as many times as needed to trigger a reconnect. 

There was also a fix made in 71940fe that could be relevant if you are not using at least version `0.6.12` (see https://github.com/eclipse/milo/issues/1199).

**Reply by pfournea** [reply]
@kevinherron migrating from 0.6.9 to 0.6.12 resolved the issue.
Thank you for helping me out!

Peter
